src/Control/Controller.php

```diff
@@ -95,6 +95,13 @@ class Controller extends RequestHandler implements TemplateGlobalProvider
         'handleIndex',
     ];
 
+    /**
+     * @config
+     *
+     * @var bool
+     */
+    private static $add_trailing_slash = false;
```

Comment:
```suggestion
     * If true, a trailing slash is added to the end of URLs, e.g. from {@link Controller::join_links()}
     */
    private static bool $add_trailing_slash = false;
```
`@config` isn't necessary, and for new API we prefer to just add explicit typehints instead of phpdoc.